[Merged by Bors] - refactor(algebra/lie/basic): split giant file into pieces

ocfnash opened the pull request:
---
Related thread [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Splitting.20up.20.60algebra.2Flie.2Fbasic.2Elean.60/near/225608592)

To make this possible to review, I have only moved things around and have resisted the temptation to add, remove, or tweak anything. The only exceptions to this is:

- I have made the variables `R` and `L` in `lie_algebra.derived_length_zero` implicit.

Aside from the fact that I tried to be careful, I am fairly confident that the above claims are accurate because following @PatrickMassot 's advice, I used the wonderful [leancrawler](https://github.com/leanprover-community/leancrawler) tool to check that things had not changed. Some details available in [this gist](https://gist.github.com/ocfnash/4de274664fd7f0e605050d06abe0a57f).

ocfnash committed:
refactor(algebra/lie/basic): split giant file into pieces

ocfnash committed:
Restore original namespacing for `is_lie_abelian_bot` and `is_solvable_of_is_nilpotent`

Identified via `leancrawler`.

ocfnash committed:
Fix long line failing lint style

bryangingechen commented on src/algebra/lie/nilpotent.lean:
```diff
@@ -0,0 +1,79 @@
+    exact lie_submodule.mono_lie _ _ _ _ h' h, },
+end
+
+/-- A Lie module is nilpotent if its lower central series reaches 0 (in a finite number of steps).-/
```
```suggestion
/-- A Lie module is nilpotent if its lower central series reaches 0 (in a finite number of
steps). -/
```

bryangingechen commented on src/algebra/lie/solvable.lean:
```diff
@@ -0,0 +1,325 @@
+
+/-- The derived length of a Lie algebra is the derived length of its 'top' Lie ideal.
+
+See also `derived_length_eq_derived_length_of_ideal`. -/
```
```suggestion
See also `lie_algebra.derived_length_eq_derived_length_of_ideal`. -/
```

bryangingechen commented on src/algebra/lie/submodule.lean:
```suggestion
/-- See also `lie_ideal.map_comap_eq`. -/
```

bryangingechen commented on src/algebra/lie/submodule.lean:
```suggestion
/-- Given two nested Lie ideals `I₁ ⊆ I₂`, the inclusion `I₁ ↪ I₂` is a morphism of Lie algebras. -/
```

bryangingechen reviewed (approved):
LGTM, thanks for taking this on!

ocfnash committed:
Update src/algebra/lie/nilpotent.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

ocfnash committed:
Update src/algebra/lie/solvable.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

ocfnash committed:
Update src/algebra/lie/submodule.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

ocfnash committed:
Update src/algebra/lie/submodule.lean

Co-authored-by: Bryan Gin-ge Chen <bryangingechen@gmail.com>

ocfnash committed:
Undo rename: keep existing name for `lie_submodule.has_bracket`

I agree with Johan that the candidate new name `lie_submodule.ideal_operation` seems less predictable.
